eclair-core/src/test/scala/fr/acinq/eclair/channel/InteractiveTxBuilderSpec.scala

```diff
@@ -0,0 +1,1083 @@
+          nodeParams.channelKeyManager.revocationPoint(channelKeyPath).publicKey,
+          nodeParams.channelKeyManager.paymentPoint(channelKeyPath).publicKey,
+          nodeParams.channelKeyManager.delayedPaymentPoint(channelKeyPath).publicKey,
+          nodeParams.channelKeyManager.htlcPoint(channelKeyPath).publicKey,
+          localParams.initFeatures,
+          None)
+    }
+
+    val firstPerCommitmentPointA = nodeParamsA.channelKeyManager.commitmentPoint(nodeParamsA.channelKeyManager.keyPath(localParamsA, ChannelConfig.standard), 0)
+    val firstPerCommitmentPointB = nodeParamsB.channelKeyManager.commitmentPoint(nodeParamsB.channelKeyManager.keyPath(localParamsB, ChannelConfig.standard), 0)
+
+    val channelId = randomBytes32()
+    val fundingScript = Script.write(Script.pay2wsh(Scripts.multiSig2of2(remoteParamsA.fundingPubKey, remoteParamsB.fundingPubKey)))
+    val fundingParamsA = InteractiveTxParams(channelId, isInitiator = true, fundingAmountA, fundingAmountB, fundingScript, lockTime, dustLimit, targetFeerate)
+    val fundingParamsB = InteractiveTxParams(channelId, isInitiator = false, fundingAmountB, fundingAmountA, fundingScript, lockTime, dustLimit, targetFeerate)
+    ChannelParams(fundingParamsA, nodeParamsA, localParamsA, remoteParamsA, firstPerCommitmentPointA, fundingParamsB, nodeParamsB, localParamsB, remoteParamsB, firstPerCommitmentPointB, channelFeatures)
+  }
+
+  case class Fixture(alice: ActorRef[InteractiveTxBuilder.Command],
+                     bob: ActorRef[InteractiveTxBuilder.Command],
+                     aliceRbf: ActorRef[InteractiveTxBuilder.Command],
+                     bobRbf: ActorRef[InteractiveTxBuilder.Command],
+                     aliceParams: InteractiveTxParams,
+                     bobParams: InteractiveTxParams,
+                     walletA: OnChainWallet,
+                     rpcClientA: BitcoinJsonRPCClient,
+                     walletB: OnChainWallet,
+                     rpcClientB: BitcoinJsonRPCClient,
+                     alice2bob: TestProbe,
+                     bob2alice: TestProbe) {
+    def forwardAlice2Bob[T <: LightningMessage](implicit t: ClassTag[T]): T = forwardMessage(alice2bob, bob)
+
+    def forwardRbfAlice2Bob[T <: LightningMessage](implicit t: ClassTag[T]): T = forwardMessage(alice2bob, bobRbf)
+
+    def forwardBob2Alice[T <: LightningMessage](implicit t: ClassTag[T]): T = forwardMessage(bob2alice, alice)
+
+    def forwardRbfBob2Alice[T <: LightningMessage](implicit t: ClassTag[T]): T = forwardMessage(bob2alice, aliceRbf)
+
+    private def forwardMessage[T <: LightningMessage](s2r: TestProbe, r: ActorRef[InteractiveTxBuilder.Command])(implicit t: ClassTag[T]): T = {
+      val msg = s2r.expectMsgType[SendMessage].msg
+      val c = t.runtimeClass.asInstanceOf[Class[T]]
+      assert(BoxedType(c).isInstance(msg), s"expected $c, found ${msg.getClass} ($msg)")
```

Comment:
What is the purpose of this? Looking at `BoxedType.apply`, that method is a no-op for non-primitive types, and `T` is contrainted to be a `LightningMessage`.

Reply:
I stole this from the internals of `expectMsgType`, as I wanted the same functionality but for an inner field (`SendMessage.msg`) so I didn't find a way of doing it with the tools available.

If you browse the source code of `expectMsgType`, you'll find this in `expectMsgClass_internal`.

Reply:
Without that code I didn't find a good way of asserting that the `msg` field is of the right generic type, but if you have a better solution I'm interested!

Reply:
I think that this is exactly equivalent:
```suggestion
      assert(c.isInstance(msg), s"expected $c, found ${msg.getClass} ($msg)")
```

Reply:
You're right! I tested this and updated it in https://github.com/ACINQ/eclair/pull/2273/commits/22a21a20ad9d120492cb29020c208926f7377689